src/PrestaShopBundle/EventListener/API/Context/ApiCartContextListener.php

```
@@ -0,0 +1,66 @@
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\EventListener\API\Context;
+
+use Cart;
+use Context;
+use Employee;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+/**
+ * Listener dedicated to set up a minimal cart/employee in legacy Context for API requests.
+ * This is needed because some legacy code (like Product::getPriceStatic) requires either
+ * an employee or a cart to be present in the context.
+ */
+class ApiCartContextListener
+{
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        if (!$event->isMainRequest()) {
+            return;
+        }
+
+        $context = Context::getContext();
+        
+        // If there's no cart in the context, create an empty one to prevent errors
+        // in legacy code that requires a cart object
+        if (!isset($context->cart) || !is_object($context->cart)) {
+            $context->cart = new Cart();
+        }
+
+        // If there's no employee in the context, create a minimal fake one
+        // This is needed for API requests where there's no actual employee session
+        if (!isset($context->employee) || !is_object($context->employee)) {
```

Hmm, no there is no meaning to an employee in the Admin API context
There is a meaning in having an ApiClient but not an employee, so the code relying on employee must be resilient to the absence of an Employee

Forcing a fake employee is ignoring the root problem

---

For the same reason, I don't think we should have an empty cart by default, it also hides some underlying problems This makes the API statefull which is something we want to avoid as much as possible

---

I'll look into it.